app/uk/gov/hmrc/transitmovementsauditing/controllers/AuditController.scala

```diff
@@ -114,6 +139,21 @@ class AuditController @Inject() (
         presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError))
       )
 
+  //TODO: combine parseObj and parseDetails??
+  //      e.g. private def parseXX[A](body: String): EitherT[Future, PresentationError, A]= ???
+  //      requires an implicit reads for the validate!
```

Comment:
Very much agree we should do this, the implicit shouldn't be an issue:

```scala
private def parse[A: Reads](body: String): EitherT[Future, PresentationError, A] = ???
```

should do the trick.